### PR TITLE
Add ITComputeTest test for resource name

### PR DESCRIPTION
**Pull request by andreamlin**
For testing https://github.com/googleapis/google-cloud-java/issues/4416.

~This test won't pass until artman is updated with https://github.com/googleapis/gapic-generator/pull/2529 and the clients are refreshed.~ Edit: blockers are gone

### CODE REVIEW

**Comment by andreamlin** [comment]
How do i format the code?

**Comment by igorbernstein2** [comment]
I think its `mvn com.coveo:fmt-maven-plugin:2.8:format`

**Comment by andreamlin** [comment]
@igorbernstein2 No luck, the tool returns success without changing any files.

**Comment by andreamlin** [comment]
@kolea2 Is there anything else I can do to fix the formatting on the diff?

**Comment by kolea2** [comment]
Hey @andreamlin! I locally ran 'mvn com.coveo:fmt-maven-plugin:format' on your branch and it reformatted the file for me - looks like there are some unused imports and it reformatted a comment. Let me know if you need help getting it to work or I can try to push my changes as well.

**Comment by andreamlin** [comment]
@kolea2  Thanks for the fix! Would you mind reviewing?

File: google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java [hunk 1]
```diff
@@ -85,6 +101,26 @@ public void testGetDiskType() {
     assertThat(diskType.getDefaultDiskSizeGb()).isNotNull();
   }
 
+  @Test
+  public void testInsertInstance() {
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java]: Do we have a test that asserts a successfully inserted instance?

File: google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java [hunk 2]
```diff
@@ -85,6 +101,26 @@ public void testGetDiskType() {
...
+    } catch (InvalidArgumentException e) {
+      // Expect a Bad Request HTTP 400 error, but it should NOT be because of a resource name
+      // problem.
+      assertThat(e.getMessage()).contains("Bad Request");
+      assertThat(e.getCause().getMessage())
+          .doesNotContain("Invalid value for field 'resource.machineType'");
+      return;
+    }
+
+    fail();
```
[the remark this answers is not in the document]
Comment (andreamlin) [reply google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java]: Done.

File: google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java [hunk 3]
```diff
@@ -85,6 +101,26 @@ public void testGetDiskType() {
...
+    Instance instance =
+        Instance.newBuilder().setName("mytestinstancetemplate").setMachineType(machineType).build();
+    try {
+      instanceClient.insertInstance(ProjectZoneName.of(DEFAULT_PROJECT, ZONE), instance);
+    } catch (InvalidArgumentException e) {
+      // Expect a Bad Request HTTP 400 error, but it should NOT be because of a resource name
+      // problem.
+      assertThat(e.getMessage()).contains("Bad Request");
+      assertThat(e.getCause().getMessage())
+          .doesNotContain("Invalid value for field 'resource.machineType'");
```
[the remark this answers is not in the document]
Comment (andreamlin) [reply google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java]: Done!

File: google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java (hunk 1 above)
Comment (andreamlin) [reply google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java]: It'll be in https://github.com/googleapis/google-cloud-java/pull/4420/files#diff-72d767a100f949c4606b04bb6820a8feR248. That is blocked on an LRO implementation tho.

**Comment by andreamlin** [comment]
Fixed up comments. PTAL

File: google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java (hunk 1 above)
Comment (andreamlin) [reply google-cloud-clients/google-cloud-compute/src/test/java/com/google/cloud/compute/v1/it/ITComputeTest.java]: So this should be fine for now, and a later PR will test inserting an instance.

**Pull request merged**